subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -950,6 +950,66 @@ static int cmd_wifi_status(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_wifi_ap_status(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_wifi_sap();
+	struct wifi_iface_status status = {0};
+
+	context.sh = sh;
+
+	if (net_mgmt(NET_REQUEST_WIFI_IFACE_STATUS, iface, &status,
+		     sizeof(struct wifi_iface_status))) {
+		PR_WARNING("Status request failed\n");
+
+		return -ENOEXEC;
+	}
+
+	uint8_t mac_string_buf[sizeof("xx:xx:xx:xx:xx:xx")];
+
+	switch (status.state) {
+	case WIFI_HAPD_IFACE_UNINITIALIZED:
+		PR("State: HAPD_IFACE_UNINITIALIZED\n");
```

Comment:
We waste flash here, so this could be replaced by
```
PR("State: %s\n", "HAPD_IFACE_UNINITIALIZED");
```
so that the format string can be shared between different case statements. Please adjust the prints below following this style.

Reply:
modified